src/v/serde/serde.h

```diff
@@ -452,6 +456,11 @@ void read_nested(iobuf_parser& in, T& t, std::size_t const bytes_left_limit) {
           read_nested<serde_size_t>(in, bytes_left_limit));
         in.consume_to(str.size(), str.begin());
         t = str;
+    } else if constexpr (std::is_same_v<Type, bytes>) {
```

Comment:
bytes is a typedef of sstring with different template parameters. what safety net exists to catch a future scenario in which types definitions change and bytes is wrongly serialized as an sstring?

Reply:
Just to clarify, is the concern here that we may one day change the type definition of `bytes` in such a way that it would be incompatible with already-serialized data? Or is the concern that sstring- and bytes- handling may inadvertently become identical? I'm curious if that's a legitimate concern -- do we frequently change the type definitions?